tea.py

```diff
@@ -0,0 +1,443 @@
+        self.stack_bd["Negative electrode electrolyte mass loading [mg.cm-2]"] = self.stack_bd.get("Negative electrode electrolyte volume loading [uL.cm-2]") * self.stack_edd.get("Electrolyte density [kg.m-3]") / 1000
+        self.stack_bd["Negative electrode inactive material mass loading [mg.cm-2]"] = self.stack_edd.get("Negative electrode density [kg.m-3]") * self.stack_edd.get("Negative electrode thickness [m]") * 100 - self.stack_bd.get("Negative electrode active material mass loading [mg.cm-2]")
+        self.stack_bd["Separator mass loading [mg.cm-2]"] = self.stack_bd.get("Separator volume loading [uL.cm-2]") * self.stack_edd.get("Separator density [kg.m-3]") / 1000
+        self.stack_bd["Separator electrolyte mass loading [mg.cm-2]"] = self.stack_bd.get("Separator electrolyte volume loading [uL.cm-2]") * self.stack_edd.get("Electrolyte density [kg.m-3]") / 1000
+        if self.parameter_values.get("Positive electrode active material volume fraction") + self.parameter_values.get("Positive electrode porosity") == 1:
+                self.stack_bd["Positive electrode active material mass loading [mg.cm-2]"] = self.stack_bd.get("Positive electrode active material volume loading [uL.cm-2]") * self.stack_edd.get("Positive electrode composite true density [kg.m-3]") / 1000
+        elif self.parameter_values.get("Positive electrode active material true density [kg.m-3]") == None:
+                self.stack_bd["Positive electrode active material mass loading [mg.cm-2]"] = 0
+                print("Error: Missing 'Positive electrode active material true density [kg.m-3]'")
+        else:
+                self.stack_bd["Positive electrode active material mass loading [mg.cm-2]"] = self.stack_bd.get("Positive electrode active material volume loading [uL.cm-2]") * self.parameter_values.get("Positive electrode active material true density [kg.m-3]") / 1000
+        self.stack_bd["Positive electrode electrolyte mass loading [mg.cm-2]"] = self.stack_bd.get("Positive electrode electrolyte volume loading [uL.cm-2]") * self.stack_edd.get("Electrolyte density [kg.m-3]") / 1000
+        self.stack_bd["Positive electrode inactive material mass loading [mg.cm-2]"] = self.stack_edd.get("Positive electrode density [kg.m-3]") * self.stack_edd.get("Positive electrode thickness [m]") * 100 - self.stack_bd.get("Positive electrode active material mass loading [mg.cm-2]")
+        self.stack_bd["Positive current collector mass loading [mg.cm-2]"] = self.stack_bd.get("Positive current collector volume loading [uL.cm-2]") * self.stack_edd.get("Positive current collector density [kg.m-3]") / 1000
+
+        return self.stack_bd
+
+    def plot_stack_breakdown(self):
+
+        if not hasattr(self, "stack_bd"):
+                self.calculate_stack_breakdown()
+
```

Comment:
here (and in other places) it might be neater to do `stack_bd = self.stack_bd ` first to avoid using `self.` everywhere